### ISSUE TITLE
Using tor binary from o/s instead of distributing it

### ISSUE THREAD

**Issue by SjonHortensius** [bisq-network/bisq#492]
As part of #484 I noticed Bitsquare distributes it's own tor binary. I think this might be a nice fallback, but I also believe Bitsquare should use the o/s binary in /usr/bin when it it available instead.

This also saves you from needing to update the tor-client whenever a new version comes out. My qnd POC can be found here: https://github.com/SjonHortensius/bitsquare/commit/338f7f117939ecc2fc302c3d57079f6c81c851a7

I'd be happy to rewrite this to something like `torPath = exists(/usr/bin/tor) ? '/usr/bin/tor' : local.path();`. It seems the best location to change this is [the OnionProxyContext constructor](https://github.com/bitsquare/bitsquare/blob/32e1a5a3accbb8ca98a0632f360ee95f1f844360/jtorproxy/src/main/java/com/msopentech/thali/toronionproxy/OnionProxyContext.java#L40)

**Comment by ManfredKarrer** [comment]
Hi @SjonHortensius, I have seen that already but I am not sure if different Tor versions work with the Java Tor libraries used in Bitsquare (JTorCtrl,...). If anyone has time to test Bitsquare with different Tor version on diff. OS we could change that. Otherwise I see it as a risk which can be easily avoided by shipping our own Tor binary.

**Comment by SjonHortensius** [comment]
@ManfredKarrer; don't you think you're also _introducing_ risk by shipping a custom binary? The tor releases seem to have a pretty stable api; since they are already used by numerous other external tools.

I understand using external dependencies gives you less control; but it also gives you some of benefits:
- auto-upgrades for security fixes that might (also) compromise Bitsquare
- no need to setup a custom compile-farm for different o/s'es and architectures
- less code to worry about / things that can go wrong

Did you already experience incompatibilities with o/s supplied tor-binaries, compared to shipping it yourself?

**Comment by ManfredKarrer** [comment]
Our Tor binaries are taken from official Tor distributions. 

I agree with your points, I just have so much other stuff on my table to do so that I dont want to spend much time to test on all OS, with all the latest Tor binaries to see if it does not break anything.

When I started with the Tor libraries I wanted to use the latest Tor binary from OSX but it was not working. I did not spend more time on investigating, so might have been a trivial issues as well. The Java Tor libs used in Bitsquare are also quite outdated and is another open task, but as said there is so much more hi-prio stuff to do that I cannot focus on that yet.

If any experienced dev will make that to his task I would be happy. The jTorCtrl lib for instance would need an update to the latest protocol APIs and the jTorProxy lib could need some love as well...
Though I don't have enough funds yet to pay for such a big task. Once we have our DAO in place I can offer easier compensation.
